apps/builder/app/builder/features/style-panel/style-source/style-source-input.tsx

```diff
@@ -191,47 +225,50 @@ const TextFieldBase: ForwardRefRenderFunction<
           aria-label="New Style Source Input"
         />
       )}
-      {value.map((item) => (
-        <StyleSourceControl
-          key={item.id}
-          menuItems={renderStyleSourceMenuItems(item)}
-          id={item.id}
-          selected={item.id === selectedItemSelector?.styleSourceId}
-          state={
-            item.id === selectedItemSelector?.styleSourceId
-              ? selectedItemSelector.state
-              : undefined
-          }
-          stateLabel={
-            item.id === selectedItemSelector?.styleSourceId
-              ? states.find(
-                  (state) => state.selector === selectedItemSelector.state
-                )?.label
-              : undefined
-          }
-          error={item.id === error?.id ? error : undefined}
-          disabled={item.disabled}
-          isDragging={item.id === dragItemId}
-          isEditing={item.id === editingItemId}
-          source={item.source}
-          onChangeEditing={(isEditing) => {
-            onEditItem?.(isEditing ? item.id : undefined);
-          }}
-          onSelect={() => onSelectItem?.({ styleSourceId: item.id })}
-          onChangeValue={(label) => {
-            onEditItem?.();
-            onChangeItem?.({ ...item, label });
-          }}
-        >
-          {item.source === "local" ? (
-            <Flex align="center" justify="center">
-              <LocalStyleIcon />
-            </Flex>
-          ) : (
-            item.label
-          )}
-        </StyleSourceControl>
-      ))}
+      {value.map((item) => {
+        return (
```

Comment:
Seems like no need in extra nesting

Reply:
John makes a great point here. 
Having collapsed panels and the states-pulldown, where you use dots to indicate “has styles” and adding a similar dot to indicate the opposite, “has no styles” makes no sense.

Simply inverting the dot-Logic in Style Sources (Local and Tags get a dot when they have styles) would, however, create a lot of noise – as the sole reason for Tokens to exist is holding data. Dots would therefore show up pretty much for any selected item. For that reason, I suggested **rather marking the exception** or error.

Placing a dot in the States dropdown, as indicated by John (see rough sketch) would clearly be an improvement – yet this required a click each time. 

![Frame 56](https://github.com/user-attachments/assets/57d26091-c21d-443d-b436-3ef6a9ad8549)

I had [originally proposed](https://discord.com/channels/955905230107738152/1323031888281075793) several other ways to indicate the Empty state – they would not clash with the symbols / logic you already use.

I also added a new, subtle option at the bottom (a little cutout).
![2024-12-31_10h55_34](https://github.com/user-attachments/assets/8c78b045-c4b1-423d-9d55-0a53bf2fbbc9)

My personal favourite is fully rounded. In a well-built project this style would **never** show for Tokens. Round would show you an error. For Local round would be the default. As it only appears on the left, it also indicates the end of the row.
![2024-12-30_08h52_25](https://github.com/user-attachments/assets/2a566be1-c90e-4b83-b44b-e45571e651b3)